cola/libtopology/topology_graph.h

```diff
@@ -239,9 +239,7 @@ namespace topology {
          * this typedef can be used to declare a wrapper functor
          * for transferStraightConstraint
          */
-        typedef std::binder1st<
-            std::mem_fun1_t<void, Segment, StraightConstraint*> 
-            > TransferStraightConstraint;
+        typedef std::function<void(StraightConstraint*)> TransferStraightConstraint;
```

Comment:
libtopology/./../libtopology/topology_graph.h|242 col 22| warning: 'template<class _Operation> class std::binder1st' is deprecated: use 'std::bind' instead [-Wdeprecated-declarations]
||   242 |         typedef std::binder1st<
||       |                      ^~~~~~~~~
|| In file included from /usr/include/c++/13/bits/stl_function.h:1438,
/usr/include/c++/13/backward/binders.h|108 col 11| note: declared here
||   108 |     class binder1st
||       |           ^~~~~~~~~
libtopology/./../libtopology/topology_graph.h|243 col 18| warning: 'template<class _Ret, class _Tp, class _Arg> class std::mem_fun1_t' is deprecated [-Wdeprecated-declarations]
||   243 |             std::mem_fun1_t<void, Segment, StraightConstraint*>
||       |                  ^~~~~~~~~~
/usr/include/c++/13/bits/stl_function.h|1300 col 11| note: declared here
||  1300 |     class mem_fun1_t : public binary_function<_Tp*, _Arg, _Ret>
||       |           ^~~~~~~~~~
|| ../libtopology/topology_graph.h: In member function 'void topology::Edge::getTopologyConstraints(std::vector<topology::TopologyConstraint*>*) const':